### PR TITLE
[12.0][MIG] account_reconcile_rule_early_payment_discount

### PR DESCRIPTION
**Pull request by p-tombez**
Migrated from :
- [ ] #210 

Depends on :
- [ ] #240 
- [ ] https://github.com/OCA/account-payment/pull/231

### CODE REVIEW

**Comment by leemannd** [comment]
The bot is lying. Cyril Gaudin had signed the cla when he did his previous PR. His comment should be ignored.

File: account_reconcile_rule_early_payment_discount/__manifest__.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+    'depends': [
+        'account_reconcile_rule',
+        'account_early_payment_discount',
+    ],
+    'website': 'https://github.com/OCA/account-reconcile',
+    'data': [
+        'views/account_reconcile_rule_view.xml',
+    ],
+    'installable': True,
+    'auto_install': True,
```
Comment (tschanzt) [inline account_reconcile_rule_early_payment_discount/__manifest__.py:19]: Why would you auto install this?

**Review by tschanzt — COMMENTED** [review]
looks good to me otherwise.

File: account_reconcile_rule_early_payment_discount/__manifest__.py (hunk 1 above)
Comment (leemannd) [reply account_reconcile_rule_early_payment_discount/__manifest__.py:19]: AFAIR this module makes the necessary glue in-between the account reconciliation and the early payment discount. And if you end-up with these two modules/functionalities installed, you want to have this one silently installed.

**Review by leemannd — APPROVED** [review]

File: account_reconcile_rule_early_payment_discount/tests/test_rule_early_payment_discount.py [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+@at_install(False)
+@post_install(True)
+class TestEarlyPaymentDiscountRule(SavepointCase):
+
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+
+        # Customer Partner
+        cls.partner = cls.env['res.partner'].with_context(
```
Comment (simahawk) [inline account_reconcile_rule_early_payment_discount/tests/test_rule_early_payment_discount.py]: pls set tracking disable at suite level `cls.env = cls.env(context=dict(cls.env.context, tracking_disable=True))`

File: account_reconcile_rule_early_payment_discount/tests/test_rule_early_payment_discount.py [hunk 3]
```diff
@@ -0,0 +1,196 @@
...
+            tracking_disable=True).create({
+                'name': 'Customer partner',
+            })
+
+        cls.rule_model = cls.env['account.reconcile.rule']
+        # Delete existing rules
+        cls.rule_model.search([]).unlink()
```
Comment (simahawk) [inline account_reconcile_rule_early_payment_discount/tests/test_rule_early_payment_discount.py]: you can use `with mute_logger('odoo.models.unlink'):` to mute the logger

File: account_reconcile_rule_early_payment_discount/tests/test_rule_early_payment_discount.py [hunk 4]
```diff
@@ -0,0 +1,196 @@
+# Copyright 2016 Cyril Gaudin (Camptocamp)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+
+from datetime import date, timedelta
+
+from odoo.tests.common import SavepointCase, at_install, post_install
+
+
+@at_install(False)
```
Comment (simahawk) [inline account_reconcile_rule_early_payment_discount/tests/test_rule_early_payment_discount.py:10]: in v12 you can use tags

**Review by simahawk — APPROVED** [review]
LG, few details to update + build is :red_circle: 
Approving to not get back as I don't see major issues.

**Comment by grindtildeath** [comment]
@p-tombez Can you cherry-pick https://github.com/OCA/account-reconcile/pull/210/commits/0a009243cd9e859e1fa75224be381a909cf8368a ?

**Comment by p-tombez** [comment]
@grindtildeath Done

**Review by grindtildeath — APPROVED** [review]
Runbot warning to be fixed on #240

**Pull request closed without merging**